text/0000-glimmer-components.md

```diff
@@ -0,0 +1,1309 @@
+  * called _after_ any child element's `{{did-render}}` modifiers
+  * called in definition order in the template
+* **May or May Not**
+  * be called in the same tick as DOM insertion
+  * have the sibling nodes fully initialized in DOM
+
+> Note: These timing semantics were mostly defined in the
+> [element modifier manager RFC](https://github.com/emberjs/rfcs/blob/master/text/0373-Element-Modifier-Managers.md)
+> and are repeated here for clarity.
+
+`{{did-render}}` receives a function with the following signature as the first
+positional parameter:
+
+```ts
+type DidRenderHandler = (element: Element, ...args): void;
+```
+
+The `element` argument is the element that the modifier is applied to, and the
+rest of the arguments are any remaining positional parameters passed to
+`{{did-render}}`. If the first positional parameter is not a callable function,
+`{{did-render}}` will throw an error. Functions which are passed to this element
+modifier will _not_ be bound to any context by default. Users can bind them
+using the `(action)` helper:
+
+```hbs
+<div {{did-render (action this.setupElement)}}>
+```
+
+#### `{{will-destroy}}`
+
+This modifier runs immediately before the element is removed. It has the
+following timing semantics:
+
+* **Always**
+  * called _after_ the enclosing component's `didUpdateArgs` hook
+  * called _before_ the enclosing component's `willDestroy` hook
+  * called _after_ any child element's `{{will-destroy}}` modifiers
+  * called in definition order in the template
+* **May or May Not**
+  * be called in the same tick as DOM insertion
+  * have the sibling nodes fully initialized in DOM
+
+> Note: These timing semantics were mostly defined in the
+> [element modifier manager RFC](https://github.com/emberjs/rfcs/blob/master/text/0373-Element-Modifier-Managers.md)
+> and are repeated here for clarity.
+
+`{{will-destroy}}` receives a function with the following signature as the first
+positional parameter:
+
+```ts
+type WillDestroyHandler = function(element: Element, ...args): void;
+```
+
+The `element` argument is the element that the modifier is applied to, and the
+rest of the arguments are any remaining positional parameters passed to
+`{{will-destroy}}`. If the first positional parameter is not a callable function,
+`{{will-destroy}}` will throw an error. Functions which are passed to this element
+modifier will _not_ be bound to any context by default. Users can bind them
+using the `(action)` helper:
+
+```hbs
+<div {{will-destroy (action this.teardownElement)}}>
+```
+
+## How we teach this
+
+Teaching Glimmer components is intrinsically tied to a wider shift in the Ember
+programming model - the Ember Octane edition. From a teaching perspective, this
+edition will be completely overhauling the guides and updating all of the best
+practices as they stand. New users should see Glimmer components as the
+_default_, and should not ever have to write a classic component or see one in
+the main guides.
```

Comment:
I think we can promise 99% of use cases being covered. The 1% case are cases where class components do things which are strictly more powerful, such as the "render-detector" use case, but these are very rare cases from what we've seen and could generally be solved in other ways (the only truly generic need to detect _any_ render was for positioning using a JS API, and ResizeObserver would have been a better solution).

I don't think we're talking about deprecating classic components just yet, it will be a _long_ time before they are deprecated. But this is a massive overhaul and rebranding effort, and we need to emphasize the new programming model. There _will_ be sections on classic usage, and they will be maintained for the forseeable future.